### PR TITLE
Add jobs for kyma-incubator/documentation-component

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Add presubmit job for [documentation-component](https://github.com/kyma-incubator/documentation-component) and for governance.
- Add related tests for new jobs.

**Related epic**
https://github.com/kyma-project/kyma/issues/4025

**Related issue(s)**
https://github.com/kyma-incubator/documentation-component/issues/13

### CODE REVIEW

File: prow/jobs/incubator/documentation-component/documentation-component.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  path_alias: github.com/kyma-incubator/documentation-component
+  max_concurrency: 10
+  extra_refs:
+    - org: kyma-project
+      repo: test-infra
+      base_ref: master
+      path_alias: github.com/kyma-project/test-infra
+  spec:
+    containers:
+      - image: eu.gcr.io/kyma-project/prow/test-infra/buildpack-node:v20181130-b28250b
```
Comment (aerfio) [inline prow/jobs/incubator/documentation-component/documentation-component.yaml]: As we have agreed to use yarn, and not node, we should make new buildpack with yarn and set it to use it here

Comment (magicmatatjahu) [reply prow/jobs/incubator/documentation-component/documentation-component.yaml]: @aerfio  So please prepare new buildpack

File: prow/config.yaml [hunk 2]
```diff
@@ -455,6 +455,10 @@ branch-protection:
               protect: true
         marketplaces:
           protect: true
+        documentation-component:
```
Comment (michal-hudy) [inline prow/config.yaml]:
```suggestion
        documentation-component:
          protect: true
```

File: prow/jobs/incubator/documentation-component/documentation-component-governance.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+presubmits: # runs on PRs
```
Comment (michal-hudy) [inline prow/jobs/incubator/documentation-component/documentation-component-governance.yaml]: Merge jobs in one file

File: prow/jobs/incubator/documentation-component/documentation-component.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          - "/home/prow/go/src/github.com/kyma-incubator/documentation-component"
+        resources:
+          requests:
+            memory: 1.5Gi
+            cpu: 0.8
+
+job_labels_template: &job_labels_template
+  preset-dind-enabled: "true"
+
+presubmits: # runs on PRs
```
Comment (michal-hudy) [inline prow/jobs/incubator/documentation-component/documentation-component.yaml]: We don't have post job?

File: prow/jobs/incubator/documentation-component/documentation-component.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  kyma-incubator/documentation-component:
+    - name: pre-master-documentation-component
```
Comment (michal-hudy) [inline prow/jobs/incubator/documentation-component/documentation-component.yaml]: Hm...maybe we can simplify naming and create it like for marketplaces repository?

File: prow/jobs/incubator/documentation-component/documentation-component.yaml (hunk 1 above)
Comment (aerfio) [reply prow/jobs/incubator/documentation-component/documentation-component.yaml]: v20190724-d41df0f

Comment (aerfio) [reply prow/jobs/incubator/documentation-component/documentation-component.yaml]:
```suggestion
      - image: eu.gcr.io/kyma-project/test-infra/buildpack-node:v20190724-d41df0f
```

**Pull request merged**